contracts/SpokePool.sol

```diff
@@ -1,16 +1,169 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.0;
 
-contract SpokePool {
-    //   constructor() {}
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@uma/core/contracts/common/implementation/Testable.sol";
+import "@uma/core/contracts/common/implementation/Lockable.sol";
+import "@uma/core/contracts/common/implementation/MultiCaller.sol";
 
+/**
+ * @title SpokePool
+ * @notice Contract deployed on source and destination chains enabling depositors to transfer assets from source to 
+ * destination. Deposit orders are fulfilled by off-chain relayers who also interact with this contract. Deposited
+ * tokens are burned on the source chain and relayers send the recipient the desired token currency and amount
+ * on the destination chain.
+ */
+contract SpokePool is Testable, Lockable, MultiCaller {
+    using SafeERC20 for IERC20;
+
+    // Chain ID for this contract.
+    uint256 public chainId;
+
+    // Timestamp when contract was constructed. Relays cannot have a quote time before this.
+    uint64 public deploymentTime;
+
+    // Track the total number of deposits. Used as a unique identifier for relays.
+    uint256 public numberOfDeposits;
+
+    // // Address of WETH on L1. If the deposited token maps to this L1 token then wrap ETH to WETH on the users behalf.
+    // address public l1Weth;
+
+    struct DestinationToken {
+        address token;
+        address spokePool;
+        uint256 chainId;
```

Comment:
having this in here has a big implication: each `SpokePool` needs to know about the waitlists of all other tokens, on all other chains. for example boba needs to know for each origin token what destination is whitelisted

Reply:
would the right data structure for `whitelistedDestinationTokens` not be:
```
mapping(address => uint256 => DestinationToken)
```

so the extra step is for the target chainID?